src/utils/mdast-to-vdom.js

```diff
@@ -215,6 +219,14 @@ class VDOMTransformer {
     // create a JSDOM object with the hast surrounded by the provided tag
     return new JSDOM(`<${tag}>${VDOMTransformer.toHTML(this._root, this._handlers)}</${tag}>`).window.document.body.firstChild;
   }
+
+  /**
+   * Resets the transformer to avoid leakages between sequential transformations
+   */
+  reset() {
```

Comment:
We need to be able to reset the transformer (i.e. the counters for the headings collision), so had to introduce a new method for it. Feel free to propose better naming if `reset` feels inappropriate